web-console/src/views/tasks-view.tsx

```diff
@@ -293,9 +298,38 @@ ORDER BY "rank" DESC, "created_time" DESC`);
     </AsyncActionDialog>;
   }
 
+  private initTableColumnSelection() {
+    if (localStorage.getItem(supervisorTableColumnSelection) == null) {
+      const columns: string[] = ["Datasource", "Type", "Topic/Stream", "Status", "Actions"];
+      const defaultSetting: any = {};
+      columns.forEach(column => defaultSetting[column] = true);
+      localStorage.setItem(supervisorTableColumnSelection, JSON.stringify(defaultSetting));
+    }
+    if (localStorage.getItem(taskTableColumnSelection) == null) {
+      const columns: string[] = ["Task ID", "Type", "Datasource", "Created time", "Status", "Duration", "Actions"];
+      const defaultSetting: any = {};
+      columns.forEach(column => defaultSetting[column] = true);
+      localStorage.setItem(taskTableColumnSelection, JSON.stringify(defaultSetting));
+    }
+  }
+
+  renderSupervisorTableColumnSelection() {
+    return <TableColumnSelection
+      onChange={() => this.setState({})}
+      tableName={supervisorTableColumnSelection}
+    />;
+  }
+
+  renderTaskTableColumnSelection() {
+    return <TableColumnSelection
+      onChange={() => this.setState({})}
+      tableName={taskTableColumnSelection}
+    />;
```

Comment:
What is the reasoning for pulling these functions out? They are so simple it might be cleaner to just inline the TableColumnSelection elements